src/storage/ducklake_transaction.cpp

```diff
@@ -104,6 +104,7 @@ struct TransactionChangeInformation {
 	set<TableIndex> dropped_views;
 	set<TableIndex> tables_inserted_into;
 	set<TableIndex> tables_deleted_from;
+	set<DataFileIndex> files_deleted_from;
```

Comment:
This could potentially grow to be a large list if we are deleting from many files. We should be able to figure out which files a transaction has deleted from by querying the tables - which doesn't require us to add more data to the change feed. Could we rework this to do that instead?

Reply:
absolutely